feat: support conditional axis + add examples (#5253)
vega/vega-lite: site/docs/encoding/axis.md

**TODO** (We have `encoding` property akin to [Vega's axis `encode`](https://vega.github.io/vega/docs/axes/#custom-axis-encodings), but within each element's block, we do not have `enter/update/exit`.)
-->

{:#conditional}

### Conditional Axis Properties

> We can set axis properties (whose type include "ConditionalAxisProperty") to a [conditional value definition](condition.html#value).

Reviewer: ```suggestion
We can set axis properties (which can be of type "ConditionalAxisProperty") to a [conditional value definition](condition.html#value).
```
